kalliope/core/OrderAnalyser.py

```diff
@@ -171,6 +187,31 @@ def _get_split_order_without_bracket(order):
         split_order = order.split()
         return split_order
 
+    @classmethod
+    def is_not_contain_matching(cls, user_order, signal_order, not_containing_words):
```

Comment:
was not already present somewhere else in the core code?

Reply:
Hmm I don't think so, this is for the new matching type, to find the not containing words.